skbio/tree/tests/test_tree.py

```diff
@@ -1303,6 +1303,42 @@ def test_shuffle_raises(self):
         with self.assertRaises(MissingNodeError):
             next(self.simple_t.shuffle(names=['x', 'y']))
 
+    def test_cache_leafcount(self):
+        t = TreeNode.read([r'((a,b)e,(c,d)f,g)r;'])
+        t._cache_leafcount()
+        self.assertEquals(t.find('r').leafcount, 5)
```

Comment:
I still don't understand how the property decorator will work.  Would we need to create stale parameters to perform the recursion if these attributes aren't filled?  Also how would be traverse up the tree if say the attributes of a leaf were accessed - but not filled?